cli/src/commands/codegen.rs

```diff
@@ -67,20 +69,52 @@ pub struct Opts {
     allow_insecure: bool,
 }
 
-fn derive_for_type_parser(src: &str) -> Result<(String, String), String> {
-    let (ty, derive) = src
-        .split_once('=')
-        .ok_or_else(|| String::from("Invalid pattern for `derive-for-type`. It should be `type=derive`, like `my_type=serde::Serialize`"))?;
+#[derive(Debug, Clone)]
+struct DeriveForType {
+    type_path: String,
+    trait_path: String,
+    recursive: bool,
+}
 
-    Ok((ty.to_string(), derive.to_string()))
+#[derive(Debug, Clone)]
+struct AttributeForType {
+    type_path: String,
+    attribute: String,
+    recursive: bool,
 }
 
-fn attributes_for_type_parser(src: &str) -> Result<(String, String), String> {
-    let (ty, attribute) = src
-        .split_once('=')
-        .ok_or_else(|| String::from("Invalid pattern for `attribute-type`. It should be `type=attribute`, like `my_type=serde::#[allow(clippy::all)]`"))?;
+fn derive_for_type_parser(src: &str) -> Result<DeriveForType, String> {
+    let (type_path, trait_path, recursive) = str_equals_str_comma_maybe_recursive_parser(src)
+    .ok_or_else(|| String::from("Invalid pattern for `derive-for-type`. It should be `type=derive` or `type=derive,recursive`, like `my_type=serde::Serialize` or `my_type=serde::Serialize,recursive`"))?;
+    Ok(DeriveForType {
+        type_path,
+        trait_path,
+        recursive,
+    })
+}
+
+fn attributes_for_type_parser(src: &str) -> Result<AttributeForType, String> {
+    let (type_path, attribute, recursive) = str_equals_str_comma_maybe_recursive_parser(src)
+    .ok_or_else(|| String::from("Invalid pattern for `attributes-for-type`. It should be `type=attribute` like `my_type=serde::#[allow(clippy::all)]` or `type=attribute,recursive` like `my_type=serde::#[allow(clippy::all)], recursive`"))?;
+    Ok(AttributeForType {
+        type_path,
+        attribute,
+        recursive,
+    })
+}
+
+/// Parses a `&str` of the form `str1=str2` into `(str1, str2, false)` or `str1=str2,recursive` into `(str1, str2, true)`.
+fn str_equals_str_comma_maybe_recursive_parser(src: &str) -> Option<(String, String, bool)> {
```

Comment:
We could probably have a few unit-tests for this. Would `str1=str2,a,b,c,recursive` also be a valid syntax?

I'd also probably rename this to `type_parser` or something similar